### ISSUE TITLE
TypeError: expected string or bytes-like object

### ISSUE THREAD

**Issue by wennycooper** [ultralytics/yolov5#5374]
- **Current repo**: run `git fetch && git status -uno` to check and `git pull` to update repo
- **Common dataset**:  coco128.yaml
- **Common environment**:  Docker image

## To Reproduce (REQUIRED)

Input:

```
# python train.py --img 640 --batch 16 --epochs 5 --data coco128.yaml --weights yolov5s.pt
```

Output:

```
Downloading https://ultralytics.com/assets/Arial.ttf to Arial.ttf...
Traceback (most recent call last):
  File "/workspace/folder/yolov5/utils/plots.py", line 56, in check_font
    return ImageFont.truetype(str(font) if font.exists() else font.name, size)
  File "/opt/conda/lib/python3.8/site-packages/PIL/ImageFont.py", line 853, in truetype
    return freetype(font)
  File "/opt/conda/lib/python3.8/site-packages/PIL/ImageFont.py", line 850, in freetype
    return FreeTypeFont(font, size, index, encoding, layout_engine)
  File "/opt/conda/lib/python3.8/site-packages/PIL/ImageFont.py", line 173, in __init__
    freetype_version = parse_version(features.version_module("freetype2"))
  File "/opt/conda/lib/python3.8/site-packages/packaging/version.py", line 49, in parse
    return Version(version)
  File "/opt/conda/lib/python3.8/site-packages/packaging/version.py", line 264, in __init__
    match = self._regex.search(version)
TypeError: expected string or bytes-like object

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "train.py", line 35, in <module>
    import val  # for end-of-epoch mAP
  File "/workspace/folder/yolov5/val.py", line 26, in <module>
    from models.experimental import attempt_load
  File "/workspace/folder/yolov5/models/experimental.py", line 10, in <module>
    from models.common import Conv
  File "/workspace/folder/yolov5/models/common.py", line 23, in <module>
    from utils.plots import Annotator, colors
  File "/workspace/folder/yolov5/utils/plots.py", line 64, in <module>
    class Annotator:
  File "/workspace/folder/yolov5/utils/plots.py", line 66, in Annotator
    check_font()  # download TTF if necessary
  File "/workspace/folder/yolov5/utils/plots.py", line 61, in check_font
    return ImageFont.truetype(str(font), size)
  File "/opt/conda/lib/python3.8/site-packages/PIL/ImageFont.py", line 853, in truetype
    return freetype(font)
  File "/opt/conda/lib/python3.8/site-packages/PIL/ImageFont.py", line 850, in freetype
    return FreeTypeFont(font, size, index, encoding, layout_engine)
  File "/opt/conda/lib/python3.8/site-packages/PIL/ImageFont.py", line 173, in __init__
    freetype_version = parse_version(features.version_module("freetype2"))
  File "/opt/conda/lib/python3.8/site-packages/packaging/version.py", line 49, in parse
    return Version(version)
  File "/opt/conda/lib/python3.8/site-packages/packaging/version.py", line 264, in __init__
    match = self._regex.search(version)
TypeError: expected string or bytes-like object
```

## Expected behavior

Expected to start training

## Environment
- OS: Ubuntu 18.04
- GPU: RTX 3060

**Comment by glenn-jocher** [comment]
@wennycooper you've got an environment error caused by PIL. Can you put a breakpoint in here to see what is being passed to ImageFont.truetype() in utils/plots.py", line 56?

```python
    return ImageFont.truetype(str(font) if font.exists() else font.name, size)
```

**Comment by wennycooper** [comment]
Sorry that I've reinstalled the environment and the problem was gone.
I will close this issue. Thanks!

**Comment by rambles-tech** [comment]
I also had this issue.  I was able to resolve it by uninstalling and then reinstalling pillow:
`pip3 uninstall -y pillow`  
`pip3 install pillow`

**Comment by glenn-jocher** [comment]
@wennycooper @rambles-tech good news 😃! Your original issue may now be fixed ✅ in PR #5592. This PR updates the Docker base image to 21.10 and also updates PIL, which was the root cause of this bug.

To receive this update:
- **[Git](https://github.com/ultralytics/yolov5)** – `git pull` from within your `yolov5/` directory or `git clone https://github.com/ultralytics/yolov5` again
- **[PyTorch Hub](https://pytorch.org/hub/ultralytics_yolov5/)** – Force-reload `model = torch.hub.load('ultralytics/yolov5', 'yolov5s', force_reload=True)`
- **[Notebooks](https://github.com/ultralytics/yolov5/blob/master/tutorial.ipynb)** – View updated notebooks  <a href="https://colab.research.google.com/github/ultralytics/yolov5/blob/master/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolov5"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **[Docker](https://hub.docker.com/r/ultralytics/yolov5)** – `sudo docker pull ultralytics/yolov5:latest` to update your image <a href="https://hub.docker.com/r/ultralytics/yolov5"><img src="https://img.shields.io/docker/pulls/ultralytics/yolov5?logo=docker" alt="Docker Pulls"></a>

Thank you for spotting this issue and informing us of the problem. Please let us know if this update resolves the issue for you, and feel free to inform us of any other **issues** you discover or **feature requests** that come to mind. Happy trainings with YOLOv5 🚀!
